src/transformers/typescript.ts

```diff
@@ -53,70 +69,161 @@ const importTransformer: ts.TransformerFactory<ts.SourceFile> = (context) => {
   return (node) => ts.visitNode(node, visit);
 };
 
-export function loadTsconfig(
-  compilerOptionsJSON: any,
-  filename: string,
-  tsOptions: Options.Typescript,
-) {
-  if (typeof tsOptions.tsconfigFile === 'boolean') {
-    return { errors: [], options: compilerOptionsJSON };
-  }
+function getComponentScriptContent(markup: string): string {
+  const regex = createTagRegex('script', 'gi');
+  let match: RegExpMatchArray;
 
-  let basePath = process.cwd();
+  while ((match = regex.exec(markup)) !== null) {
+    const { context } = parseAttributes(match[1]);
 
-  const fileDirectory = (tsOptions.tsconfigDirectory ||
-    dirname(filename)) as string;
+    if (context !== 'module') {
+      return match[2];
+    }
+  }
 
-  let tsconfigFile =
-    tsOptions.tsconfigFile ||
-    ts.findConfigFile(fileDirectory, ts.sys.fileExists);
+  return '';
+}
 
-  if (!tsconfigFile) {
-    return { errors: [], options: {} };
+function createSourceMapChain({
+  filename,
+  content,
+  compilerOptions,
+}: {
+  filename: string;
+  content: string;
+  compilerOptions: CompilerOptions;
+}): SourceMapChain | undefined {
+  if (compilerOptions.sourceMap) {
+    return {
+      content: {
+        [filename]: content,
+      },
+      sourcemaps: {},
+    };
   }
+}
 
-  tsconfigFile = isAbsolute(tsconfigFile)
-    ? tsconfigFile
-    : join(basePath, tsconfigFile);
+function injectVarsToCode({
+  content,
+  markup,
+  filename,
+  attributes,
+  sourceMapChain,
+}: {
+  content: string;
+  markup?: string;
+  filename: string;
+  attributes?: Record<string, any>;
+  sourceMapChain?: SourceMapChain;
+}): string {
+  if (!markup) return content;
+
+  const { vars } = compile(stripTags(markup), {
+    generate: false,
+    varsReport: 'full',
+    errorMode: 'warn',
+    filename,
+  });
 
-  basePath = dirname(tsconfigFile);
+  const sep = '\nconst $$$$$$$$ = null;\n';
+  const varsValues = vars.map((v) => v.name).join(',');
+  const injectedVars = `const $$vars$$ = [${varsValues}];`;
```

Comment:
Who knows, maybe some unknown globals/values are used in the markup, and we don't want to add unnecessary diagnostics
```suggestion
  const injectedVars = `// @ts-ignore
const $$vars$$ = [${varsValues}];`;
```

Reply:
@dummdidumm, this is not necessary. Since we are only transpiling the code, there is no type checking...

Reply:
You are right, the only diagnostics that are returned are syntax errors which do not happen in that area. Undone.